app/models/order_status.rb

```diff
@@ -122,6 +122,7 @@ def non_protected_statuses(facility)
       new_status.self_and_descendants
         .or(in_process.self_and_descendants)
         .or(canceled.self_and_descendants)
+        .or(unrecoverable.self_and_descendants)
```

Comment:
Why is this change needed? I see this method is used in a few places and I don't know if there could be any unintended side effects

Reply:
On the OrderDetailsController I added the unrecoverable status to the eligible statuses for completed orders so that you can change it to unrecoverable. This is not enough, once the order is set as unrecoverable, if you open the order details view, the unrecoverable status is not visible as it's not a completed order anymore, so it appears with the "new" status. This is why I needed to add the unrecoverable status here too.

Reply:
In addition, the controllers use this non_protected_statuses method to set the available status options for an order. We have logic order_detail.rb that defines the permissions needed for an order to be updated to a certain status so that if the Unrecoverable status appears as an option for an order that can't be set as unrecoverable and the user tries to do so, the app won't allow it and an error message will show up.